### PR TITLE
Guard window/document references with is-browser checks

### PR DESCRIPTION
**Pull request by adamburmister**
To: @zackargyle 

Added:
- Add `isBrowser` const to files referencing window/document
- Guard methods where server side rendering doesn't make sense

Tested:
- Current client side rendering continues to work

Coming in future PRs:
- Testing framework or example which will demonstrate isomorphic rendering

### CODE REVIEW

File: src/components/PinItButton.js [hunk 1]
```diff
@@ -80,15 +83,21 @@ export default class PinItButton extends PinterestBase {
...
+              url = url || window.document.URL;
+              description = description || window.document.title;
+            }
+
             href = Const.URL.PIN_CREATE + `?guid=${Config.guid}`;
             href += `&media=${encodeURIComponent(media)}`;
-            href += `&url=${encodeURIComponent(url || document.URL)}`;
-            href += `&description=${encodeURIComponent(description || document.title)}`;
+            href += `&url=${encodeURIComponent(url)}`;
+            href += `&description=${encodeURIComponent(description)}`;
```
Comment (zackargyle) [inline src/components/PinItButton.js:100]: Both of these functions are click handlers, so I don't think we need any non-browser checks here. Am I right?

Comment (adamburmister) [inline src/components/PinItButton.js:100]: Yeah, but wouldn't this URL be pre-rendered on a server side render as it's passed via props to the template?

Comment (zackargyle) [inline src/components/PinItButton.js:100]: Oh shoot, I did not even pay attention, this is not a click-handler anyway. Yup. Good call.

Comment (adamburmister) [inline src/components/PinItButton.js:100]: Resolved :white_check_mark:

**Pull request merged**